packages/serialize/src/lib/Serialize.ts

```diff
@@ -8,34 +8,48 @@ export namespace Serialize {
-    else if (typeof data === 'number') return { type: Type.Number, value: data };
-    else if (isObject(data)) return { type: Type.Object, value: toJSONObject(data) };
-    else if (data instanceof RegExp) return { type: Type.RegExp, value: { source: data.source, flags: data.flags } };
-    else if (data instanceof Set) return { type: Type.Set, value: toJSONArray(Array.from(data)) };
-    else if (typeof data === 'string') return { type: Type.String, value: data };
-    else if (data === undefined) return { type: Type.Undefined, value: 'undefined' };
+    if (Array.isArray(data)) return { [Keying.Type]: Type.Array, [Keying.Value]: toJSONArray(data) };
+    else if (typeof data === 'bigint') return { [Keying.Type]: Type.BigInt, [Keying.Value]: data.toString() };
+    else if (typeof data === 'boolean') return { [Keying.Type]: Type.Boolean, [Keying.Value]: data };
+    else if (data instanceof Date) return { [Keying.Type]: Type.Date, [Keying.Value]: data.toJSON() };
+    else if (data instanceof Map) return { [Keying.Type]: Type.Map, [Keying.Value]: toJSONEntries(Array.from(data)) };
+    else if (data === null) return { [Keying.Type]: Type.Null, [Keying.Value]: null };
+    else if (typeof data === 'number') return { [Keying.Type]: Type.Number, [Keying.Value]: data };
+    else if (isObject(data)) return { [Keying.Type]: Type.Object, [Keying.Value]: toJSONObject(data) };
+    else if (data instanceof RegExp) return { [Keying.Type]: Type.RegExp, [Keying.Value]: { s: data.source, f: data.flags } };
+    else if (data instanceof Set) return { [Keying.Type]: Type.Set, [Keying.Value]: toJSONArray(Array.from(data)) };
+    else if (typeof data === 'string') return { [Keying.Type]: Type.String, [Keying.Value]: data };
+    else if (data === undefined) return { [Keying.Type]: Type.Undefined, [Keying.Value]: 'undefined' };
 
     throw new TypeError(
       `Serialize received an unknown type while formatting: "${data.constructor.name}", see @joshdb/transform for custom serialization`
     );
   }
 
   function toJSONArray(array: unknown[]): JSON[] {
-    return array.reduce<JSON[]>((json, value) => [...json, toJSON(value)], []);
+    const json: JSON[] = [];
+
+    for (const value of array) json.push(toJSON(value));
+    return json;
   }
 
   function toJSONEntries(entries: [PropertyKey, unknown][]): [PropertyKey, JSON][] {
-    return entries.reduce<[PropertyKey, JSON][]>((json, [key, value]) => [...json, [key, toJSON(value)]], []);
+    const json: [PropertyKey, JSON][] = [];
+
+    for (const [key, value] of entries) json.push([key, toJSON(value)]);
+    return json;
   }
 
   function toJSONObject(object: Record<PropertyKey, unknown>): Record<PropertyKey, JSON> {
-    return Object.entries(object).reduce<Record<PropertyKey, JSON>>((json, [key, value]) => ({ ...json, [key]: toJSON(value) }), {});
+    const json: Record<PropertyKey, JSON> = {};
+
+    for (const key in object) {
+      if ({}.hasOwnProperty.call(object, key)) {
```

Comment:
```suggestion
      if (object.hasOwnProperty(key)) {
```

Reply:
Don't think this works, if you want it can be

Object.prototype.hasOwnProperty.call

Reply:
![image](https://user-images.githubusercontent.com/46537907/201226063-8a466869-9377-4ced-be0e-f5ebc4fd3b62.png)